.github/workflows/scripts/upload_reports/upload.py

```diff
@@ -0,0 +1,118 @@
+from argparse import ArgumentParser, Namespace
+import requests
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from pathlib import Path
+import json
+import os
+
+
+class PLOSSSettings(BaseSettings):
+    """Settings for the PennyLane OSS service."""
+
+    model_config = SettingsConfigDict(env_prefix="PENNYLANE_OSS_SERVER_")
+
+    endpoint_url: str
+    """URL of the PennyLane OSS Service."""
+
+    api_key: str
+    """API key to be supplied to the PennyLane OSS Service."""
+
+
+def read_reports() -> dict:
+    """Read the reports from the reports directory."""
+
+    # Look for reports in the absolute path where GitHub Actions writes them
+    github_workspace = os.environ.get("GITHUB_WORKSPACE", "")
```

Comment:
Any concern with usage of env-vars for something like this? Can these be safely pulled, and not overriden by users/3rd party devs with PRs against us?

Reply:
A valid concern! I believe the risks here are accessing keys via forked PRs or through workflow modifications. We should be covered for since we're using secrets and passing them to workflows securely.

Specifically for `GITHUB_WORKSPACE`, this is a standard GitHub-provided environment variable that simply indicates the path where your repository is checked out so is pretty harmless anyway. Nonetheless, it can't be overridden by PR authors in a way that would compromise security given the above points.

@rashidnhm would be good if you can verify!